File: Java/ID1.java
/*
ID: 1
If we list all the natural numbers below 10 that are multiples of 3 or 5, we get 3, 5, 6 and 9. The sum of these multiples is 23. 
Find the sum of all the multiples of 3 or 5 below 1000.
*/
public class ID1 {

	public static void main(String[] args) {
		System.out.println(SumDivisibleBy(3) + SumDivisibleBy(5) - SumDivisibleBy(15));
	}

	public static int SumDivisibleBy(int factor) {
		int sum = 0;
		for (int i = 0; i <= 999; i++) {
			int p = 999 / factor;
			sum = factor * (p * (p + 1)) / 2;
		}
		return sum;
	}

}
// Answer: 233168

File: R/ID1.R
#ID: 1
#If we list all the natural numbers below 10 that are multiples of 3 or 5, we get 3, 5, 6 and 9. The sum of these multiples is 23. 
#Find the sum of all the multiples of 3 or 5 below 100.

sum.of.multiples.3.5.below <- function(max){
p <- seq(1, (max-1))
multiples.of.3 <- num[p %% 3 == 0]
multiples.of.5 <- num[p %% 5 == 0]
num.request <- union(multiples.of.3, multiples.of.5)
answer <- sum(num.request)
return(answer)
}
sum.of.multiples.3.5.below(1000)

#Answer: 233168
